src/StateService/Storage/StateRoot.cs

```diff
@@ -0,0 +1,131 @@
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Network.P2P;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using System;
+using System.IO;
+
+namespace Neo.Plugins.StateService.Storage
+{
+    public class StateRoot : ICloneable<StateRoot>, IVerifiable, ISerializable
+    {
+        public byte Version;
+        public uint Index;
+        public UInt256 RootHash;
+        public Witness Witness;
+
+        private UInt256 _hash = null;
+
+        public UInt256 Hash
+        {
+            get
+            {
+                if (_hash is null)
+                {
+                    _hash = new UInt256(Crypto.Hash256(this.GetHashData()));
+                }
+                return _hash;
+            }
+        }
+
+        Witness[] IVerifiable.Witnesses
+        {
+            get
+            {
+                return new[] { Witness };
+            }
+            set
+            {
+                if (value.Length != 1) throw new ArgumentException();
+                Witness = value[0];
+            }
+        }
+
+        public int Size =>
+            sizeof(byte) +      //Version
+            sizeof(uint) +      //Index
+            UInt256.Length +    //RootHash
+            1 +                 //Witness array count
+            Witness.Size;       //Witness
```

Comment:
In case `Witness` is `null`, an Exception will be thrown here.

Reply:
1

Reply:
Done.